tests/test_fragments.py

```diff
@@ -0,0 +1,82 @@
+import tiledb
+from common import test_array_names
+from commands.root import root
+
+import itertools
+import numpy as np
+import os
+import pytest
+import shutil
+import sys
+
+
+@pytest.fixture(scope="function")
+def uri(temp_rootdir):
+    """
+    Create a simple dense test array.
+    """
+    path = os.path.abspath(os.path.join(temp_rootdir, "test_array"))
+
+    ctx = tiledb.default_ctx()
+    rows_dim = tiledb.Dim(ctx=ctx, domain=(1, 25), dtype=np.int64)
+    cols_dim = tiledb.Dim(ctx=ctx, domain=(1, 12), dtype=np.int64)
+    dom = tiledb.Domain(rows_dim, cols_dim, ctx=ctx)
+    att1 = tiledb.Attr(name="a", ctx=ctx, dtype=np.float64)
+    att2 = tiledb.Attr(name="b", ctx=ctx, dtype=np.float64)
+    schema = tiledb.ArraySchema(ctx=ctx, domain=dom, attrs=(att1, att2))
+
+    tiledb.DenseArray.create(path, schema)
+
+    data = np.reshape(np.arange(300), (25, 12))
+
+    for ts in range(1, 4):
+        with tiledb.DenseArray(path, mode="w", timestamp=ts) as A:
+            A[:] = {"a": data, "b": data}
+
+    yield path
+
+    shutil.rmtree(path)
```

Comment:
I thought that pytest was supposed to clean this up automatically?

Reply:
No I don't think so. Even in TileDB-Py, there's a [teardown function](https://github.com/TileDB-Inc/TileDB-Py/blob/dev/tiledb/tests/common.py#L75) that calls rmtree to clean up everything.